File: 이상욱/BOJ/m2w2/BOJ1874.java
package m2w7;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class BOJ1874 {
	
	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		
		int[] input = new int[n];
		
		for(int i = 0; i < n; i++) {
			
			input[i] = Integer.parseInt(br.readLine());
			
		}
		
		int num = 1;
		int idx = 0;
		Stack<Integer> stk = new Stack<>();
		
		StringBuilder sb = new StringBuilder();
		
		while(idx < n) {
			
			if(num > n) {
				System.out.println("NO");
				return;
			}
			
			stk.push(num);
			sb.append("+\n");
			
			while(!stk.isEmpty() && stk.peek() == input[idx]) {
				
				stk.pop();
				sb.append("-\n");
				idx++;
			}
			
			num++;
		}
		
		System.out.println(sb.toString());
		
	}
}

File: 이주은/BOJ_Kotlin/7주차/BOJ_1874.kt

import java.io.BufferedReader
import java.io.InputStreamReader
import java.lang.StringBuilder
import java.util.*

fun main()=with(BufferedReader(InputStreamReader(System.`in`))){

    val n = readLine().toInt()
    var stack = Stack<Int>()
    var idx = 0
    var sb = StringBuilder()

    for(i in 0 until n){
        var num = readLine().toInt()
        if (num > idx){
            for(j in idx + 1 .. num){
                stack.push(j)
                sb.append("+\n")
            }
            idx = num
        }

        else if(stack.peek() != num){
            println("NO")
            return
        }

        stack.pop()
        sb.append("-\n")
    }
    print(sb)
}